dmutils/dates.py

```diff
@@ -0,0 +1,54 @@
+from six import string_types
+from datetime import datetime, timedelta
+from workdays import workday
+
+
+def get_publishing_dates(brief):
+    APPLICATION_OPEN_DAYS = {'1 week': 7, '2 weeks': 14}
+    QUESTIONS_OPEN_DAYS = {'1 week': 2, '2 weeks': 5}
+    dates = {}
+
+    if _brief_is_from_api_(brief):
+        dates['published_date'] = _get_start_date_from_brief_api(brief)
+    elif _brief_is_from_frontend_app_and_published_(brief):
+        dates['published_date'] = _get_start_date_from_published_frontend_brief(brief)
+    else:
+        dates['published_date'] = _get_todays_date()
+
+    length = _get_length_of_brief(brief)
+
+    dates['closing_date'] = dates['published_date'] + timedelta(days=APPLICATION_OPEN_DAYS[length])
+    dates['questions_close'] = workday(dates['published_date'], QUESTIONS_OPEN_DAYS[length])
+    dates['answers_close'] = workday(dates['closing_date'], -1)
+    dates['application_open_weeks'] = APPLICATION_OPEN_DAYS[length]//7
```

Comment:
This is similar to the point above, if a brief doesn't have a requirementLength, like a non specialist, then setting this key an using it in the template makes sense. If requirementsLenght was called on a brief without it, the template would be off.